### ISSUE TITLE
BIG-IP 15.1.5.1 Always Random "msg": "No JSON object could be decoded"

### ISSUE THREAD

**Issue by Fuhunter** [F5Networks/f5-ansible#2214]
##### COMPONENT NAME

bigip_management_route
bigip_snmp
...
basically every module. It is not possible to pin it down to a single module

##### ANSIBLE VERSION

```
ansible 2.9.25
  config file = /app/ansible/install_and_config_server/ansible.cfg
  configured module search path = [u'/home/ansible/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug 13 2020, 02:51:10) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```
##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     15.1.5.1
  Build       0.0.14
  Edition     Point Release 1
  Date        Sun Mar 27 23:23:42 PDT 2022
```
##### CONFIGURATION

ansible.cfg changes:
callback_whitelist = profile_tasks

##### OS / ENVIRONMENT

Ansible host is RHEL 7.9

##### SUMMARY

After upgrading or F5 from 13.1.5 to 15.1.5.1 our Ansible Rollout process isn't working anymore. It affects all of our roles which we are using for basic configuration of our server.
It doesn't matter if we use a singe role with a single task or multiple roles at once, it is always the same error at different points while executing the playbook. It may happen while executing the very first task, or when executing the 3rd one etc..

We get the following error:

 "msg": "No JSON object could be decoded"

With Version 13.1.5 wie didn't have that problem.

##### STEPS TO REPRODUCE

```
- name: Create a management route dnslog
  bigip_management_route:
    name: syslog
    description: Route to dnslog
    gateway: 192.168.1.1
    network: 192.168.1.2/32
    state: present
    provider:
      password: "{{ ansible_user_password }}"
      server: "{{ inventory_hostname }}"
      user: "{{ ansible_user }}"
      validate_certs: no
  delegate_to: localhost
```

##### EXPECTED RESULTS

The task should complete without an error.

##### ACTUAL RESULTS

Tasks fails with  "msg": "No JSON object could be decoded"

```
<localhost> Using network group action bigip for bigip_management_route
<localhost> connection transport is rest
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: xxxx
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp `"&& mkdir "` echo $HOME/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923 `" && echo ansible-tmp-1652883384.38-32224-142203260851923="` echo $HOME/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/network/f5/bigip_management_route.py
<localhost> PUT /home/ansible/.ansible/tmp/ansible-local-31893jqkDzC/tmpQZzRLO TO /home/ansible/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923/AnsiballZ_bigip_management_route.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/ansible/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923/ /home/ansible/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923/AnsiballZ_bigip_management_route.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/python2 /home/ansible/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923/AnsiballZ_bigip_management_route.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /home/ansible/.ansible/tmp/ansible-tmp-1652883384.38-32224-142203260851923/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_bigip_management_route_payload_fB31Eu/ansible_bigip_management_route_payload.zip/ansible/modules/network/f5/bigip_management_route.py", line 446, in main
  File "/tmp/ansible_bigip_management_route_payload_fB31Eu/ansible_bigip_management_route_payload.zip/ansible/modules/network/f5/bigip_management_route.py", line 272, in exec_module
  File "/tmp/ansible_bigip_management_route_payload_fB31Eu/ansible_bigip_management_route_payload.zip/ansible/modules/network/f5/bigip_management_route.py", line 293, in present
  File "/tmp/ansible_bigip_management_route_payload_fB31Eu/ansible_bigip_management_route_payload.zip/ansible/modules/network/f5/bigip_management_route.py", line 303, in update
  File "/tmp/ansible_bigip_management_route_payload_fB31Eu/ansible_bigip_management_route_payload.zip/ansible/modules/network/f5/bigip_management_route.py", line 404, in read_current_from_device
fatal: [blnglv-test.lanbf.db.de -> localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "description": "Route to dnslog",
            "gateway": "192.168.1.1",
            "name": "dnslog",
            "network": "192.168.1.2/32",
            "partition": "Common",
            "provider": {
                "auth_provider": null,
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
                "server": "NAME",
                "server_port": null,
                "ssh_keyfile": null,
                "timeout": null,
                "transport": "rest",
                "user": "xxxx",
                "validate_certs": false
            },
            "state": "present"
        }
    },
    "msg": "No JSON object could be decoded"
}
```

I am not sure but does this may have something todo with these known issues?

853161-4 : Restjavad has different behavior for error responses if the body is over 2k
538283-2 : iControl REST asynchronous tasks may block other tasks from running

**Comment by Cajga** [comment]
We have the same issue here with 14.1.4.6! Seems this is a regression introduced with the latest fix to the Critical CVE as the issue started right after the upgrade from 14.1.4.1.
We use exclusively ansible to configure our F5 clusters and this bug basically prevents us to work properly.

I found that if I redo the task then I may get it working after some tries:
```
$ ANSIBLE_ENABLE_TASK_DEBUGGER=True ansible-playbook -i inventory/hosts --limit prod_ext01 --tag data_groups,irules,virtual_servers playbook.yaml
Vault password:
.
.
.
TASK [Collect BIG-IP facts] ******************************************************************************************************************************************************************************************************************
fatal: [lbx01.example.com]: FAILED! => {"changed": false, "msg": "No JSON object could be decoded"}
[lbx01.example.com] TASK: Collect BIG-IP facts (debug)> redo
ok: [lbh01.example.com]
ok: [lbh02.example.com]
fatal: [lbx01.example.com]: FAILED! => {"changed": false, "msg": "No JSON object could be decoded"}
[lbx01.example.com] TASK: Collect BIG-IP facts (debug)> redo
ok: [lbx02.example.com]
fatal: [lbx01.example.com]: FAILED! => {"changed": false, "msg": "No JSON object could be decoded"}
[lbx01.example.com] TASK: Collect BIG-IP facts (debug)> redo
ok: [lbx01.example.com]
```

**Comment by trinaths** [comment]
Created [INFRAANO-761] for internal tracking.

**Comment by mstrent** [comment]
We just upgraded to 14.1.4.6 and started hitting this as well.

I believe it's related to this KB and bug:
https://support.f5.com/csp/article/K91334123
https://cdn.f5.com/product/bugtracker/ID1108181.html

**Comment by Cajga** [comment]
In our case, it is not related to this bug. We did a long debug session with an F5 engineer which resulted in the following facts:
* the failing request cannot be seen in the tcpdump of the management interface (after decryption of https traffic) on the F5 system. When our ansible looped through resources (like pool_members) we saw the request before and after the failing one but not the failing one itself. This is also true for the logs of the REST API.
* in ansible logs, we saw the request going out so it "disappeared" between the ansible server and the REST API server

As a next step, we want to capture the tcpdump on the ansible server and decrypt it. Will post here the results.
